### PR TITLE
Show assignment stats and performance report based on student period at the time of assignment

### PR DESCRIPTION
**Pull request by Dantemss**
# Done
- Added period attribute to taskings
- Populate period when tasks are distributed
- Use new period attribute in assignment stats
- Use new period attribute in performance report

### CODE REVIEW

**Comment by Dantemss** [comment]
Merge https://github.com/openstax/tutor-server/pull/490 first

**Comment by Dantemss** [comment]
The removed lines refer to cassettes that were combined into 1, in the tasked_task_plan factory.

**Comment by Dantemss** [comment]
@jpslav ready

File: app/models/entity/role.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Entity::Role < Tutor::SubSystems::BaseModel
   enum role_type: [:unassigned, :default, :teacher, :student]
 
-  has_many :students, dependent: :destroy, subsystem: :course_membership
-  has_many :teachers, dependent: :destroy, subsystem: :course_membership
+  has_one :student, dependent: :destroy, subsystem: :course_membership
```
Comment (jpslav) [inline app/models/entity/role.rb]: gracias

**Comment by jpslav** [comment]
Thanks @Dantemss!

**Comment by jpslav** [comment]
Actually one thing missing is that we need to set the period ID in the taskings for practice widgets, so that those get counted in the learning guide, etc.

**Pull request merged**